app/controllers/application_controller/web_authenticator.rb

```diff
@@ -0,0 +1,17 @@
+class ApplicationController
+  class WebAuthenticator < Struct.new(:controller)
+    def current_user
+      @current_user ||= User.where(id: controller.session[:user_id]).first
```

Comment:
Here I need the controller just so I can look at its session hash. Maybe I should just initialize with the session